src/codegen/sys/functions.rs

```diff
@@ -183,9 +208,64 @@ fn generate_object_funcs(
         )?;
     }
     if write_get_type {
-        let configured_functions = obj.functions.matched(&glib_get_type);
-        generate_cfg_configure(w, &configured_functions, false)?;
-        writeln!(w, "    pub fn {}() -> GType;", glib_get_type)?;
+        let namespace = &env.namespaces[library::MAIN_NAMESPACE];
+
+        let symbol_prefix = if symbol_prefix == INTERN {
+            use crate::case::CaseExt;
+
+            let symbol_prefix = c_type.to_snake();
+            assert!(
+                glib_get_type.starts_with(&symbol_prefix),
+                "Can't guess symbol prefix for type {} with get_type() function {}",
+                c_type,
+                glib_get_type
+            );
+            namespace
+                .symbol_prefixes
+                .iter()
+                .filter_map(|prefix| {
+                    if symbol_prefix.starts_with(prefix) {
+                        Some(String::from(&symbol_prefix[(prefix.len() + 1)..]))
+                    } else {
+                        None
+                    }
+                })
+                .next()
+                .unwrap_or_else(|| {
+                    panic!(
+                        "Can't guess symbol prefix for type {} with get_type() function {}",
```

Comment:
Can you make this message slightly different from previous ?

Reply:
Actually that makes no sense at all, let me fix that properly :)

Reply:
Done, all working a bit different now